tests/unit/metadata/test_single_table.py

```diff
@@ -872,39 +867,68 @@ def test_set_primary_key_validation(self):
         # Setup
         instance = SingleTableMetadata()
 
-        err_msg = "'primary_key' must be a string or tuple of strings."
+        err_msg = (
+            "'primary_key' must be a string or tuple of strings."
+        )
         # Run / Assert
         with pytest.raises(ValueError, match=err_msg):
             instance.set_primary_key(('1', 2, '3'))
 
+    def test_set_primary_key_validation_columns(self):
+        """Test that ``set_primary_key`` crashes for invalid arguments.
+
+        Setup:
+            - A ``SingleTableMetadata`` instance with ``_columns`` set.
+
+        Input:
+            - A tuple with columns not present in ``_columns``.
+
+        Side Effect:
+            - A ``ValueError`` should be raised.
+        """
+        # Setup
+        instance = SingleTableMetadata()
+        instance._columns = {'a', 'd'}
+
+        err_msg = (
+            "Unknown primary key values {'b'}."
+            ' Keys should be columns that exist in the table.'
+        )
+        # Run / Assert
+        with pytest.raises(ValueError, match=err_msg):
+            instance.set_primary_key(('a', 'b', 'd'))
+            # NOTE: used to be ('a', 'b', 'd', 'c')
```

Comment:
what are these notes for?

Reply:
For Plamen to know where it used to be a set of errors, to change back after we decide how to handle them.